nautobot/core/api/serializers.py

```diff
@@ -233,6 +338,45 @@ def build_field(self, field_name, info, model_class, nested_depth):
 
         return super().build_field(field_name, info, model_class, nested_depth)
 
+    def _get_natural_key_lookups_value_for_field(self, field_name, natural_key_field_instance):
+        """Extract natural key field lookups for a specific field name.
+
+        Args:
+            field_name (str): The field name to extract lookups for.
+            natural_key_field_instance (dict): The dict containing natural key field values.
+        """
+        data = {}
+        for key, value in natural_key_field_instance.items():
+            if key.startswith(f"{field_name}__"):
+                if isinstance(value, uuid.UUID):
+                    value = str(value)
+                elif value == "ObjectNotFound":
+                    continue
+                elif not value:
+                    data[key] = CSV_NON_TYPE
+                else:
+                    data[key] = value
+        return data
+
+    def to_representation(self, instance):
+        data = super().to_representation(instance)
+        altered_data = {}
+
+        if self._is_csv_request() and self.natural_keys_values:
```

Comment:
Since `natural_keys_values` is defaulted to `None` this is more dogmatic.

```suggestion
        if self._is_csv_request() and self.natural_keys_values is not None:
```